modules/serverless-inmemorychannel.adoc

```diff
@@ -0,0 +1,66 @@
+// Module included in the following assemblies:
+//
+//  * serverless/knative_eventing/serverless-channels.adoc
+
+[id="serverless-inmemorychannel_{context}"]
+= Using the default InMemoryChannel configuration
+
+InMemoryChannels are for development use only, and should not be used in a production environment.
+
+The following are limitations of InMemoryChannel channels:
+
+* No event persistence is available. If a Pod goes down, events on that Pod will be lost.
+* InMemoryChannel channels do not implement event ordering, so two events that are received in the channel at the same time may be delivered to a subscriber in any order.
+* If a subscriber rejects an event, there are no redelivery attempts. Instead, the rejected event is sent to a _dead letter sink_ if this sink exists, or is otherwise dropped.
+
+When you install Knative Eventing, the following custom resource definition is created automatically:
+
+[source,yaml]
+----
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  namespace: knative-eventing
+  name: config-br-default-channel
+data:
+  channelTemplateSpec: |
+    apiVersion: messaging.knative.dev/v1
+    kind: InMemoryChannel
+----
+// TODO: Add link for dead letter sink docs once these are added.
+
+.Creating a channel using the cluster default configuration
+
+* Create a generic `Channel` custom object.
++
+[source,yaml]
+----
+apiVersion: messaging.knative.dev/v1beta1
+kind: Channel
+metadata:
+  name: example-channel
+  namespace: default
+----
++
+When the above `Channel` object is created, a mutating admission webhook adds a set of `spec.channelTemplate` properties for the `Channel` object based on the default channel implementation.
++
+[source,yaml]
+----
+apiVersion: messaging.knative.dev/v1beta1
+kind: Channel
+metadata:
+  name: example-channel
+  namespace: default
+spec:
+  channelTemplate:
+    apiVersion: messaging.knative.dev/v1beta1
+    kind: InMemoryChannel
+----
+
+The channel controller will then create the backing channel instance based on that `spec.channelTemplate` configuration. The `spec.channelTemplate` properties cannot be changed after creation, because they are set by the default channel mechanism rather than by the user.
+
+When this mechanism is used, two objects are created, a generic channel, and an InMemoryChannel channel.
+
+The generic channel acts as a proxy that copies its subscriptions to the InMemoryChannel, and sets its status to reflect the status of the backing InMemoryChannel channel.
+
+Because the channel in this example is created in the default namespace, the channel uses the cluster default, which is InMemoryChannel.
```

Comment:
I'd recommend capitalizing `Channel` in this paragraphs as well -- similar to how we do with `InMemoryChannel`